File: sgi-eti-service/src/main/java/org/crue/hercules/sgi/eti/model/Formulario.java
package org.crue.hercules.sgi.eti.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * Formulario
 */

@Entity
@Table(name = "formulario")
@Data
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class Formulario extends BaseEntity {

  public enum Tipo {
    /** Memoria */
    MEMORIA,
    /** Seguimiento Anual */
    SEGUIMIENTO_ANUAL,
    /** Seguimiento Final */
    SEGUIMIENTO_FINAL,
    /** Retrospectiva */
    RETROSPECTIVA;
  }

  /** Titulo de la documentación de Seguimiento Anual */
  public enum SeguimientoAnualDocumentacionTitle {
    /** Memoria */
    TITULO_1,
    /** Seguimiento Anual */
    TITULO_2,
    /** Seguimiento Final */
    TITULO_3;
  }

  /**
   * Serial version
   */
  private static final long serialVersionUID = 1L;

  /** Id. */
  @Id
  @Column(name = "id", length = 28, nullable = false)
  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "formulario_seq")
  @SequenceGenerator(name = "formulario_seq", sequenceName = "formulario_seq", allocationSize = 1)
  private Long id;

  /** Tipo */
  @Column(name = "tipo", nullable = false)
  @Enumerated(EnumType.STRING)
  private Tipo tipo;

  /** Codigo */
  @Column(name = "codigo", nullable = false, unique = true, length = 50)
  private String codigo;

  /** Titulo de la documentación del seguimiento anual */
  @Column(name = "seguimiento_anual_documentacion_title", nullable = false)
  @Enumerated(EnumType.STRING)
  private SeguimientoAnualDocumentacionTitle seguimientoAnualDocumentacionTitle;

}

File: sgi-webapp/src/app/core/models/eti/formulario.ts
import { IMemoria } from './memoria';
import { TIPO_EVALUACION } from './tipo-evaluacion';


export function resolveFormularioByTipoEvaluacionAndComite(tipoEvaluacion: TIPO_EVALUACION, memoria: IMemoria): IFormulario {
  switch (tipoEvaluacion) {
    case TIPO_EVALUACION.MEMORIA:
      return memoria.formulario;
    case TIPO_EVALUACION.RETROSPECTIVA:
      return memoria.formularioRetrospectiva;
    case TIPO_EVALUACION.SEGUIMIENTO_ANUAL:
      return memoria.formularioSeguimientoAnual;
    case TIPO_EVALUACION.SEGUIMIENTO_FINAL:
      return memoria.formularioSeguimientoFinal;
    default:
      return null;
  }
}

export interface IFormulario {
  /** Id */
  id: number;
  /** Tipo */
  tipo: FormularioTipo;
  /** Codigo */
  codigo: string;
  /** Titulo del apartado Seguimiento Anual en la documentación de la Memoria */
  seguimientoAnualDocumentacionTitle: FormularioSeguimientoAnualDocumentacionTitle;
  /** Activo */
  activo: boolean;
}

export enum FormularioTipo {
  MEMORIA = 'MEMORIA',
  SEGUIMIENTO_ANUAL = 'SEGUIMIENTO_ANUAL',
  SEGUIMIENTO_FINAL = 'SEGUIMIENTO_FINAL',
  RETROSPECTIVA = 'RETROSPTECTIVA'
}

export enum FormularioSeguimientoAnualDocumentacionTitle {
  TITULO_1 = 'TITULO_1',
  TITULO_2 = 'TITULO_2',
  TITULO_3 = 'TITULO_3'
}